tests/test_derivatives.py

```diff
@@ -639,13 +639,22 @@ def test_zero_spec(self):
         """
         grid = Grid((11, 11))
         x, y = grid.dimensions
-        f = Function(name="f", grid=grid, space_order=4)
+        f = Function(name="f", grid=grid, space_order=2)
         # Check that both specifications match
         drv0 = Derivative(f, (x, 2))
         drv1 = Derivative(f, (x, 2), (y, 0))
-        assert drv0.dims == drv1.dims
-        assert drv0.fd_order == drv1.fd_order
-        assert drv0.deriv_order == drv1.deriv_order
+        assert drv0.dims == (x,)
+        assert drv1.dims == (x, y)
+        assert drv0.fd_order == 2
+        assert drv1.fd_order == (2, 2)
+        assert drv0.deriv_order == 2
+        assert drv1.deriv_order == (2, 0)
+
+        assert drv0.evaluate == drv1.evaluate
+        fmean = .5 * (f + f._subs(y, y + y.spacing))
+        drv1x0 = drv1(x0={y: y+y.spacing/2}).evaluate
+
+        assert simplify(fmean.dx2.evaluate - drv1x0) == 0
```

Comment:
It would also be a good idea to check equivalence between the `Derivative(f, y, deriv_order=0, x0=y+y.spacing/2)` and `Derivative(f, (y, 0), x0={y: y+y.spacing/2})` specifications.

There should also be tests for `Derivative(f, (x, 0), (y, 0), x0={x: x+x.spacing/2, y: y+y.spacing/2})` and `Derivative(f, x, y,, deriv_order=(0, 0), x0={x: x+x.spacing/2, y: y+y.spacing/2})`

Reply:
Added